File: DSA_practice/Others/SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {
        int [] a = new int[]{2, 5, 1, 6, 15, 90, 7};
        System.out.println("Array before sorting");
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i]+" ");
        }
        System.out.println("\nArray after sorting");
        sort(a, 7);
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i]+" ");
        }
    }
    public static void sort(int [] arr, int n){
        for (int i = 0; i < n - 1; i++) {
            int mini = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[mini]) {
                    mini = j;
                }
            }
            //swap
            int temp = arr[mini];
            arr[mini] = arr[i];
            arr[i] = temp;
        }
    }
}

File: Algorithms_Analysis/selectionSort.c
#include <stdio.h>
int main()
{
    int arr[] = {2, 3, 1, 9, 12, 4, 0};
    printf("Before Sorting\n");
    for (int i = 0; i < 7; i++)
    {
        printf("%d ", arr[i]);
    }

    // Sorting
    for (int i = 0; i < 7 - 1; i++)
    {
        int minIndex = i;
        for (int j = i + 1; j < 7; j++)
        {
            if (arr[j] < arr[minIndex])
                minIndex = j;
        }
        int temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
    }

    printf("\nAfter Sorting\n");
    for (int i = 0; i < 7; i++)
    {
        printf("%d ", arr[i]);
    }
    return 0;
}